File: R/ggspec.R
#' Translate ggplot2 object into a ggspec
#'
#' @param plt A ggplot2 object.
#'
#' @return `list`, a ggspec list.
#' @export
#'
#' @examples
#' library(ggplot2)
#' p <- ggplot(data = iris)
#' p <- p + geom_point(aes(x = Petal.Width, y = Petal.Length))
#' ggspec(p)
ggspec <- function(plt){
  int_data <- data_int(plt$data, plt$layers)

  list(
    data = data_spc(int_data),
    layers = layer_spc(plt$layers, int_data),
    scales = scale_spc(plt$scales$scales),
    labels = plt$labels
  )
}

File: src-ext/Typescript/ggschema/src/ggSpec.ts
import {Datasets} from './data';
import {Layers} from './layer';
import {Labels} from './labels';
import {Scale} from './scale';

export * from './data';
export * from './layer';
export * from './labels';
export * from './scale';

export interface TopLevelSpec {
  data: Datasets;
  layers: Layers;
  scales: Scale[];
  labels: Labels;
}
